mlrun/track/tracker.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import typing
+
+from mlrun.execution import MLClientCtx
+
+
+class Tracker(abc.ABC):
+    """
+    general tracker class, includes basic demands for tracker classes,
+    in order to log 3rd party vendor's artifacts into MLRun
+    '
+    """
+
+    MODULE_NAME = ...
+
+    def __init__(self):
+        self._utils = None
+        self._tracked_platform = None  # assuming only one is being used every time
+        self._client = None
+
+    def utils(self):
+        if self._utils:
+            return self._utils
+        from mlrun.frameworks._common import (
+            CommonUtils,
+        )  # needed to avoid import issues later
+
+        self._utils = CommonUtils.convert_np_dtype_to_value_type
+        return self._utils
```

Comment:
create a `utils.py` that will have the zipping, schema to feature and this function as well.